tools/jvm/src/main/scala/org/scalajs/core/tools/optimizer/ScalaJSClosureOptimizer.scala

```diff
@@ -151,34 +167,224 @@ object ScalaJSClosureOptimizer {
+
+    def withCustomOutputWrapper(customOutputWrapper: (String, String)): Config =
+      copyWith(customOutputWrapper = customOutputWrapper)
+
+    @deprecated("Not a case class anymore", "0.6.5")
+    def copy(
+        output: WritableVirtualJSFile = this.output,
+        cache: Option[WritableVirtualTextFile] = this.cache,
+        bypassLinkingErrors: Boolean = this.bypassLinkingErrors,
+        checkIR: Boolean = this.checkIR,
+        unCache: Boolean = this.unCache,
+        disableOptimizer: Boolean = this.disableOptimizer,
+        batchMode: Boolean = this.batchMode,
+        wantSourceMap: Boolean = this.wantSourceMap,
+        prettyPrint: Boolean = this.prettyPrint,
+        relativizeSourceMapBase: Option[URI] = this.relativizeSourceMapBase,
+        noWarnMissing: Seq[ScalaJSOptimizer.NoWarnMissing] = this.noWarnMissing): Config = {
+
+      copyWith(output, cache, bypassLinkingErrors, checkIR, unCache,
+          disableOptimizer, batchMode, wantSourceMap, prettyPrint,
+          relativizeSourceMapBase, noWarnMissing)
+    }
+
+    private def copyWith(
+        output: WritableVirtualJSFile = this.output,
+        cache: Option[WritableVirtualTextFile] = this.cache,
+        bypassLinkingErrors: Boolean = this.bypassLinkingErrors,
+        checkIR: Boolean = this.checkIR,
+        unCache: Boolean = this.unCache,
+        disableOptimizer: Boolean = this.disableOptimizer,
+        batchMode: Boolean = this.batchMode,
+        wantSourceMap: Boolean = this.wantSourceMap,
+        prettyPrint: Boolean = this.prettyPrint,
+        relativizeSourceMapBase: Option[URI] = this.relativizeSourceMapBase,
+        noWarnMissing: Seq[ScalaJSOptimizer.NoWarnMissing] = this.noWarnMissing,
+        customOutputWrapper: (String, String) = this.customOutputWrapper): Config = {
+
+      new Config(output, cache, bypassLinkingErrors, checkIR, unCache,
+          disableOptimizer, batchMode, wantSourceMap, prettyPrint,
+          relativizeSourceMapBase, noWarnMissing, customOutputWrapper)
+    }
+
+    // For binary compatibility
+    @deprecated("Not a case class anymore", "0.6.5")
+    def canEqual(that: Any): Boolean = true
+
+    // For binary compatibility
+    @deprecated("Not a case class anymore", "0.6.5")
+    def productArity: Int = productArray.length
+
+    // For binary compatibility
+    @deprecated("Not a case class anymore", "0.6.5")
+    def productElement(n: Int): Any = productArray(n)
+
+    private def productArray: Array[Any] = {
+      Array[Any](output, cache, bypassLinkingErrors, checkIR, unCache,
+          disableOptimizer, batchMode, wantSourceMap, prettyPrint,
+          relativizeSourceMapBase, noWarnMissing, customOutputWrapper)
+    }
+
+    // For binary compatibility
+    override def equals(other: Any): Boolean = super.equals(other)
+
+    // For binary compatibility
+    override def hashCode(): Int = super.hashCode()
+
+    // For binary compatibility
+    override def toString(): String = super.toString()
```

Comment:
Did you test that these still behave the same way? (I.e. value by value equality rather than object equality?)